src/components/Tabs/Tabs.tsx

```diff
@@ -12,15 +13,45 @@ interface ITabsProps {
 }
 
 export const Tabs = ({ children }: ITabsProps) => {
+  const [showScrollButton, setShowScrollButton] = useState(false);
+  const tabContent = document.getElementById('tab-content');
+
+  const onResize = useCallback(() => {
+    tabContent && setShowScrollButton(tabContent.scrollWidth > window.innerWidth);
+  }, [tabContent]);
+
+  window.addEventListener('resize', onResize);
```

Comment:
This should be executed inside an useEffect lest the event handler is not assigned on each render.
```
useEffect(() => {
  window.addEventListener('resize', onResize);
}, [onResize]);
````

Although,  `const tabContent = document.getElementById('tab-content');` needs to be moved inside `onResize` callback.